### PR TITLE
Offload short range in CoulombPBCAA

### PR DESCRIPTION
**Pull request by ye-luo**
Review after #3839

## Proposed changes
Offload short range in CoulombPBCAA. The particle range will be chopped into chunks and then compute the pair distance and short range Coulomb chunk by chunk. The reason doing is is to avoid storing AA full table on the device while a large enough chunk size should ensure heavy enough workload per offload kernel. Hard-coded size 64 should be reasonable but will need further investigation.

## What type(s) of changes does this code introduce?
- New feature

### Does this introduce a breaking change?
- No

## What systems has this change been tested on?
epyc-server

## Checklist
- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- Yes. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- Yes. Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by ye-luo** [comment]
Test this please

**Comment by ye-luo** [comment]
Test this please

**Comment by ye-luo** [comment]
Test this please

**Comment by ye-luo** [comment]
Test this please

**Review by prckent — COMMENTED** [review]
Two very minor questions:
1. I noticed a couple of pragma unrolls. Did you see any benefit when adding these? In 2022 I would hope that the compilers can make a sensible choice for a small fixed size loop.
1. Unless codecov is confused, the coverage on the change is a little low. Are you able to easily expand the coverage to get >44%?

**Comment by prckent** [comment]
Test this please

**Review by PDoakORNL — APPROVED** [review]
This looks looks pretty good for me.

I made a couple of suggestions but they are just nice to have.

**Comment by ye-luo** [comment]
> Two very minor questions:
> 
>     1. I noticed a couple of pragma unrolls. Did you see any benefit when adding these? In 2022 I would hope that the compilers can make a sensible choice for a small fixed size loop.

computeDist is simplified from computeDistancesOffload. So unroll is kept. Need to assess both places to determine the removal.

> 
>     2. Unless codecov is confused, the coverage on the change is a little low. Are you able to easily expand the coverage to get >44%?

Need to run code coverage with offload to host.

File: src/QMCHamiltonians/tests/test_coulomb_pbcAA_ewald.cpp [hunk 1]
```diff
@@ -51,7 +51,7 @@ TEST_CASE("Coulomb PBC A-A Ewald3D", "[hamiltonian]")
   LRCoulombSingleton::CoulombHandler->initBreakup(ions);
 
 
-  CoulombPBCAA caa = CoulombPBCAA(ions, false);
+  CoulombPBCAA caa(ions, false, false, false);
```
[the remark this answers is not in the document]
Comment (ye-luo) [reply src/QMCHamiltonians/tests/test_coulomb_pbcAA_ewald.cpp:54]: Good suggestion. I saw computeForces labelled as deprecated. I think the list won't grow further.

**Pull request merged**